tests/TestCase/src/Controller/BlogPostsControllerTest.php

```diff
@@ -29,41 +32,43 @@ public function setUp(): void
 
     public function testBlogPostDetailOnlinePublicLoggedOut()
     {
-        $this->httpClient->get($this->Slug->getBlogPostDetail(2, 'Demo Blog Artikel'));
-        $this->assert200OkHeader();
+        $_SERVER['REQUEST_URI'] = $this->Slug->getBlogPostDetail(2, 'Demo Blog Artikel');
```

Comment:
$_SERVER['REQUEST_URI'] should not be set in test case. Is there a better solution to stop MenuHelper complaining?

Reply:
I see multiple other options:
1. define `$_SERVER['REQUEST_URI']` in `phpunit.xml` see `<server>` element in https://phpunit.readthedocs.io/en/9.0/configuration.html#the-php-element
2. in the MenuHelper check if code is run from phpunit similar to `header.php` https://github.com/foodcoopshop/foodcoopshop/pull/529/files#diff-dd7e8b4e9539e4296688523beebb8916R22
3. hide the warnings

I think the question is what we want to test, if the MenuHelper is part of the test we should set it to something realistic (mean leave it as it is), if not we can go for one of the other possibilities

Reply:
I think option 1 is good for us. MenuHelper is not part of the tests. Thanks!